### PR TITLE
Move search to toplevel in accordionlist

### LINKED ISSUE #41085: Make search surface columns when I’m working on joined tables
[Product doc](https://www.notion.so/metabase/Make-search-surface-columns-when-I-m-working-on-joined-tables-bc79b831564a4db2a646ab7a70b86580)

### Goal
Make searching on joined data in filters etc. more usable by surfacing column names as well.

- [ ] Update `AccordionList` to allow searching of all sections at once
- [ ] Keep supporting the previous usecases of AccordionList

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/41085

Moves the search box in the accordion list to the top level.

Screenshot 2024-03-12 at 17 51 17

Screenshot 2024-04-01 at 12 56 52

### CODE REVIEW

**Review by cdeweyx — APPROVED** [review]

**Review by AlessioLaiso — APPROVED** [review]

File: frontend/src/metabase/querying/components/FilterPicker/FilterColumnPicker/FilterColumnPicker.tsx [hunk 1]
```diff
@@ -29,16 +29,20 @@ export interface FilterColumnPickerProps {
...
+  alwaysVisible?: boolean;
 };
 
 const CUSTOM_EXPRESSION_SECTION: Section = {
   key: "custom-expression",
+  type: "action",
   name: t`Custom Expression`,
   items: [],
   icon: "filter",
+  alwaysVisible: true,
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterPicker/FilterColumnPicker/FilterColumnPicker.tsx:45]: I understand that currently `alwaysVisible` will be `true` only when `type` is `"action"`, so this information is equivalent.
Do we plan to allow for different permutations?
If not, WDYT about dropping `alwaysVisible` and assuming sections of type `"action"` are always visible?

File: frontend/src/metabase/core/components/AccordionList/AccordionList.jsx [hunk 2]
```diff
@@ -62,6 +62,7 @@ export default class AccordionList extends Component {
     sections: PropTypes.array.isRequired,
 
     initiallyOpenSection: PropTypes.number,
+    globalSearch: PropTypes.bool,
```
Comment (kamilmielnik) [inline frontend/src/metabase/core/components/AccordionList/AccordionList.jsx:65]: In Admin -> Table metadata -> Semantic type input, when searching, section names will no longer be displayed. Is this expected?

See [`master`](https://github.com/metabase/metabase/assets/6830683/b8e51e46-28d0-4060-9de5-9bd48dda9d50) vs [this PR](https://github.com/metabase/metabase/assets/6830683/631eb015-2c21-4909-b5a4-79d3d9358f10)

**Review by kamilmielnik — COMMENTED** [review]
Wow, this `AccordionList` component looks... adventurous.
Good job finding your way in!

We could use Mantine for the new section type, but given the state of this component I think it's ok to skip it.

I left some comments with possibly unexpected differences in behavior that I found - maybe they're expected, but it's better to double-check.
I believe 1 test has been updated incorrectly.

Can you please link relevant issue/epic/Slack link in the PR description?

File: frontend/src/metabase/core/components/AccordionList/AccordionList.jsx (hunk 2 above)
Comment (romeovs) [reply frontend/src/metabase/core/components/AccordionList/AccordionList.jsx:65]: Good catch: that was a bug. It now works as expected:

Screenshot 2024-04-04 at 17 03 07


Screenshot 2024-04-04 at 17 04 01

File: frontend/src/metabase/core/components/AccordionList/AccordionList.jsx [hunk 3]
```diff
@@ -480,6 +498,18 @@ export default class AccordionList extends Component {
     }
 
     if (globalSearch) {
+      const isSearching = searchText.length > 0;
+      const isEmpty = rows.filter(x => x.type === "item").length === 0;
```
Comment (kamilmielnik) [inline frontend/src/metabase/core/components/AccordionList/AccordionList.jsx:502]:
```suggestion
      const isEmpty = rows.filter(row => row.type === "item").length === 0;
```

File: frontend/src/metabase/admin/datamodel/metadata/components/SemanticTypeAndTargetPicker/SemanticTypeAndTargetPicker.tsx [hunk 4]
```diff
@@ -111,6 +111,7 @@ const SemanticTypeAndTargetPicker = ({
         optionSectionFn={getTypeOptionSection}
         placeholder={t`Select a semantic type`}
         searchProp="name"
+        globalSearch
```
Comment (kamilmielnik) [inline frontend/src/metabase/admin/datamodel/metadata/components/SemanticTypeAndTargetPicker/SemanticTypeAndTargetPicker.tsx:114]: `globalSearch` reminds me of search in the app header or in the command palette.

How about `allSectionsSearch`?

**Review by kamilmielnik — APPROVED** [review]
Looks good, works well :+1: 
Approving although I'm not sure about the status of this project (is this PoC?).

@romeovs Can you please link relevant issue/epic/Slack/docs in the PR description?

File: frontend/src/metabase/core/components/AccordionList/AccordionList.jsx [hunk 5]
```diff
@@ -104,6 +105,7 @@ export default class AccordionList extends Component {
   static defaultProps = {
     style: {},
     width: 300,
+    globalSearch: false,
```
[the remark this answers is not in the document]
Comment (romeovs) [reply frontend/src/metabase/core/components/AccordionList/AccordionList.jsx:108]: @AlessioLaiso what do you think?

[the remark this answers is not in the document]
Comment (AlessioLaiso) [reply frontend/src/metabase/core/components/AccordionList/AccordionList.jsx:108]: Romeo and I just caught up on this and yes, let's add dividers where needed, like here. This matches the new designs for custom expressions too, where the various expression shortcuts are separated by a divider from the "custom expression" option.

**Pull request merged**